libs/web3-api-provider/src/webb-provider/vanchor-actions.ts

```diff
@@ -171,8 +171,13 @@ export class Web3VAnchorActions extends VAnchorActions<
       // If no relayer is set, then the fee is 0, otherwise it is the fee amount
       const feeVal = relayer === ZERO_ADDRESS ? ZERO_BIG_INT : feeAmount;
       const refund = relayer === ZERO_ADDRESS ? ZERO_BIG_INT : refundAmount;
+
       const recipient =
-        relayer === ZERO_ADDRESS ? ZERO_ADDRESS : refundRecipient;
+        relayer === ZERO_ADDRESS
```

Comment:
I don't understand why to select the recipient we first check if the relayer is the zero address. As it currently stands the recipient could be set to non-zero, but if relayer is zero, you would set the recipient to zero as well.

Reply:
When the relayer is a zero address, it implies that users have opted not to use a relayer, thus both the refund amount and the recipient (in this case, the refund recipient address) are also set to zero addresses. This statement can be rephrased as follows:

```ts
const recipient =
  relayer === ZERO_ADDRESS || refundRecipient.length === 0
    ? ZERO_ADDRESS
    : refundRecipient;
```